game/src/graphic/textures/TextureHandler.java

```diff
@@ -2,30 +2,67 @@
 
 import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.files.FileHandle;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Arrays;
+import java.util.Comparator;
 import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.Predicate;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 /**
  * A texture handler class for managing textures paths for further use.
  *
- * <p>Singleton (possibly thread-safe).
+ * <p>This class is designed as Singleton, because all asset paths should be read only once at the
+ * beginning of the application.
  */
 public class TextureHandler {
+    public static final String PLACEHOLDER_FILENAME = ".resource_root";
+
     private static final TextureHandler INSTANCE = new TextureHandler();
 
     private final Map<String, Set<FileHandle>> pathMap = new LinkedHashMap<>();
 
     private TextureHandler() {
-        addAllAssets(Gdx.files.internal(Gdx.files.getLocalStoragePath()));
+        try {
+            addAllAssets(getResourceRoot());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Ich würde diese Exception hier nicht catchen, sondern an den Aufrufer weitergeben. Der kann ggf. entscheiden, was damit passieren soll. 

Außerdem würde ich nicht die aussagekräftige `IOException` durch eine nichtssagende `RuntimeException` ersetzen. Was ist der Sinn dahinter?

Reply:
@cagix Es gibt hier imo nur zwei Möglichkeiten, a) die IOE in eine schwächere RE umzuwandeln oder b) die Anwendung (mit einer Fehlermeldung) zu beenden?

Reply:
Vielleicht ist es ja auch eine nicht so gute Idee, das hier mit dem Statischen Konstruktor zu machen.

Reply:
singleton lazy initialization ginge auch... dann hätte man den static-Block nicht mehr, aber die Frage, wie mit der IOException geschickt umgehen, bleibt

Reply:
Wie wäre es mit im psvm in Game? Dort werden ja auch alle Konfigurationen eingelesen, warum sollte dies dort nicht auch hinpassen? Dort könnte dann ein Log vom Error gemacht werden und gar nicht erst der Launcher gestartet werden.

Reply:
graphic.textures.TextureHandler#getInstance wird an mehreren Stellen aufgerufen, das ginge in der main nicht

Reply:
Also hier wäre meine Idee wie ich das jetzt lösen würde. 

Variante 1 (Error eager):
1. Eine Methode initialize welche die instanz erstellt. 
2. Diese Methode im psvm aufrufen

Variante 2 (Error lazy): 
1. getInstance den static konstruktor code geben und nur aufrufen wenn instance == null
2. Im Hero beim erstellen der Animationen die IOException weiterreichen nach oben
3. Im AnimationBuilder die IOException weiterreichen nach oben reichen
4. Exception im Game abfangen und zum beenden des Programmes nutzen

Reply:
Das soll doch zum Teil in #587 gemacht werden...